src/Utilities/DoubleDicts.jl

```diff
@@ -147,11 +155,19 @@ function Base.get(
     key::MOI.ConstraintIndex{F,S},
     default,
 ) where {F,S}
-    inner = getindex(d, F, S)
-    if haskey(inner, key)
-        return inner[key]
+    inner = d[F, S]
+    return get(inner, key, default)
+end
+
+function Base.get(
+    d::AbstractDoubleDictInner,
+    key::MOI.ConstraintIndex{F,S},
+    default,
+) where {F,S}
+    if !haskey(d, key)
```

Comment:
why do we need haskey? cant you do it with get?

AbstractDoubleDictInner already has parameters F and S, they should be in the function signature.
You can have a second method for non-matching F,S

Reply:
If we used
```julia
x = get(d.inner, key.value, default)
```
we wouldn't know whether to return `typed_value(x)` or `x`.

Reply:
cant you define `typed_value` to be a no-op if the return type is correct?